modules/checkbox/react/spec/Checkbox.spec.tsx

```diff
@@ -1,102 +1,104 @@
-      .getDOMNode();
-    expect(input.getAttribute('data-propspread')).toBe('test');
-    component.unmount();
+  describe('when rendered with disabled attribute', () => {
+    it('should render a disabled checkbox input', () => {
+      const {getByRole} = render(<Checkbox disabled={true} onChange={cb} />);
+      expect(getByRole('checkbox')).toHaveProperty('disabled', true);
+    });
   });
-});
 
-describe('Checkbox Accessibility', () => {
-  test('Checkbox should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <Checkbox id={'123'} label={'Label'} onChange={jest.fn()} />
-    );
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when rendered without an id', () => {
+    it('should create a unique id for each instance', async () => {
+      const {getByLabelText} = render(
+        <form>
+          <Checkbox checked={true} onChange={cb} disabled={false} label="label1" />;
+          <Checkbox onChange={cb} disabled={false} label="label2" />;
+        </form>
+      );
+
+      const id1 = getByLabelText('label1').getAttribute('id');
+      const id2 = getByLabelText('label2').getAttribute('id');
+
+      expect(id1).not.toEqual(id2);
+    });
+
+    it('should keep the same unique id if re-rendered', () => {
+      const {getByRole, rerender} = render(<Checkbox checked={false} onChange={cb} />);
+
+      const uniqueId = getByRole('checkbox').getAttribute('id');
+      expect(getByRole('checkbox')).toHaveProperty('id', uniqueId);
+
+      rerender(<Checkbox checked={true} onChange={cb} />);
+
+      expect(getByRole('checkbox')).toHaveProperty('checked');
+      expect(getByRole('checkbox')).toHaveProperty('id', uniqueId);
+    });
   });
 
-  test('Checkbox without a defined id should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(<Checkbox label={'Label'} onChange={jest.fn()} />);
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when rendered with extra, arbitrary props', () => {
+    it('should spread extra props onto the checkbox', () => {
+      const attr = 'test';
+      const {getByRole} = render(<Checkbox data-propspread={attr} onChange={cb} />);
+      expect(getByRole('checkbox')).toHaveAttribute('data-propspread', attr);
+    });
   });
 
-  test('Checkbox wrapped in a FormField should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <FormField label="My Field" inputId="my-checkbox-field">
-        <Checkbox disabled={false} checked={true} id="my-checkbox-field" onChange={jest.fn()} />;
-      </FormField>
-    );
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when rendered with an input ref', () => {
+    it('input ref should be defined', () => {
+      const ref = React.createRef<HTMLInputElement>();
+      render(<Checkbox inputRef={ref} onChange={cb} />);
+      expect(ref.current).toBeDefined();
```

Comment:
While, this test does verify the spec `'input ref should be defined'`, shouldn't there be another spec (or adjust the wording on this one) that verifies whether or not `current` refers to the checkbox input itself? Something like `expect(inputRef.current).toHaveAttribute('role', 'checkbox');`

Reply:
After going through this same thing in #386 this will always pass since React sets the value to `null`. It may be good to have two assertions: 1) the current property is not null and 2) the current property refers to the checkbox